app/src/test/java/com/github/se/travelpouch/model/ListTravelViewModelUnitTest.kt

```diff
@@ -0,0 +1,216 @@
+package com.github.se.travelpouch.model
+
+import com.google.firebase.Timestamp
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Assert.assertNotEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.*
+import org.mockito.kotlin.anyOrNull
+import org.mockito.kotlin.whenever
+
+@ExperimentalCoroutinesApi
+class ListTravelViewModelTest {
+
+  private lateinit var travelRepository: TravelRepository
+  private lateinit var listTravelViewModel: ListTravelViewModel
+
+  private val travel =
+      TravelContainer(
+          "6NU2zp2oGdA34s1Q1q5h",
+          "Test Title",
+          "Test Description",
+          Timestamp.now(),
+          Timestamp(Timestamp.now().seconds + 1000, 0),
+          Location(
+              0.0,
+              0.0,
+              Timestamp.now(),
+              "Test Location",
+          ),
+          mapOf("Test Key item" to "Test Value item"),
+          mapOf(Participant("SGzOL8yn0JmAVaTdvG9v") to Role.OWNER))
+
+  @Before
+  fun setUp() {
+    travelRepository = mock(TravelRepository::class.java)
+    listTravelViewModel = ListTravelViewModel(travelRepository)
+  }
+
+  @Test
+  fun getNewUid_returnsUniqueId() {
+    whenever(travelRepository.getNewUid()).thenReturn("6NU2zp2oGdA34s1Q1q5h")
+    assertThat(listTravelViewModel.getNewUid(), `is`("6NU2zp2oGdA34s1Q1q5h"))
+  }
+
+  @Test
+  fun getNewUid_returnsDifferentIdsOnSubsequentCalls() {
+    val firstUid = "uniqueId123"
+    val secondUid = "uniqueId456"
+    whenever(travelRepository.getNewUid()).thenReturn(firstUid, secondUid)
+
+    val firstCallUid = listTravelViewModel.getNewUid()
+    val secondCallUid = listTravelViewModel.getNewUid()
+
+    assertNotEquals(firstCallUid, secondCallUid)
+    verify(travelRepository, times(2)).getNewUid()
+  }
+
+  @Test
+  fun getTravels_successfulFetch_updatesTravels() {
+    val travelList = listOf(travel)
+    doAnswer { invocation ->
```

Comment:
You’re repeating similar mock setups for multiple tests. Consider extracting these into helper functions to avoid duplication.